src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -332,16 +343,38 @@ public void Dispose()
             {
                 if (!this.IsDisposed)
                 {
-                    this.acceptanceSource.TrySetCanceled();
-                    this.optionsAppliedTaskSource?.TrySetCanceled();
-
                     PipeWriter? mxStreamIOWriter;
                     lock (this.SyncObject)
                     {
                         this.isDisposed = true;
                         mxStreamIOWriter = this.mxStreamIOWriter;
                     }
 
+                    TraceSource traceSrc = this.GetTraceSource();
+                    if (traceSrc.Switch.ShouldTrace(TraceEventType.Information))
+                    {
+                        traceSrc.TraceEvent(
+                            TraceEventType.Information,
+                            (int)TraceEventId.WriteError,
+                            "Calling TrySetCanceled of acceptanceSource in Dispose method of channel {0}",
```

Comment:
Why are we tracing out this message without even checking whether an error occurred?